torch/testing/_comparison.py

```diff
@@ -1077,6 +1081,7 @@ def assert_close(
     Raises:
         ValueError: If no :class:`torch.Tensor` can be constructed from an input.
         ValueError: If only ``rtol`` or ``atol`` is specified.
+        ValueError: If a tensor is a meta tensor. This is a temporary restriction and will be relaxed in the future.
```

Comment:
Should this be a notimpl error, one that will cause a test using meta tensors to skip itself?

Reply:
This is only the patch for the public `assert_close`. Here I would prefer `ValueError`, since `NotImplementedError` is usually used to indicate abstract methods that need to be overwritten by a user. Since there is nothing for the user to do here, I feel like this could be misleading.

Still, you are right, that this needs to be `NotImplementedError` in our internal test suite, since they can be turned into `unittest.SkipTest`'s. #67796 uses a small wrapper to change the `ValueError` back to a `NotImplementedError` keeping the custom error message.

Reply:
On a second thought, I could also live with a [`NotImplementedError`](https://docs.python.org/3/library/exceptions.html#NotImplementedError):

> In user defined base classes, abstract methods should raise this exception when they require derived classes to override the method, or while the class is being developed to **indicate that the real implementation still needs to be added.** [emphasis mine]

We could use it for all temporary restrictions for which the final judgement is still out.

Reply:
Yeah, I think NotImplementedError is reasonable -- it indicates the functionality is not implemented.